src/chainstate/stacks/boot/pox_2_tests.rs

```diff
@@ -0,0 +1,322 @@
+                    vrf_proof,
+                    tip.total_burn,
+                    microblock_pubkeyhash,
+                )
+                .unwrap();
+                let (anchored_block, _size, _cost) =
+                    StacksBlockBuilder::make_anchored_block_from_txs(
+                        block_builder,
+                        chainstate,
+                        &sortdb.index_conn(),
+                        block_txs,
+                    )
+                    .unwrap();
+                (anchored_block, vec![])
+            },
+        );
+
+        let (_, _, consensus_hash) = peer.next_burnchain_block(burn_ops);
+        peer.process_stacks_epoch_at_tip(&stacks_block, &microblocks);
+
+        let total_liquid_ustx = get_liquid_ustx(&mut peer);
+        let tip_index_block =
+            StacksBlockHeader::make_index_block_hash(&consensus_hash, &stacks_block.block_hash());
+
+        eprintln!("tenure_id: {}", tenure_id);
+        let alice_balance = get_balance(&mut peer, &key_to_stacks_addr(&alice).into());
+
+        let expected_alice_balance = if tenure_id < 1 {
+            1024 * POX_THRESHOLD_STEPS_USTX
+        } else if tenure_id < AUTO_UNLOCK_HT {
+            0
+        } else if tenure_id < 13 {
+            1024 * POX_THRESHOLD_STEPS_USTX
+        } else {
+            512 * POX_THRESHOLD_STEPS_USTX
+        };
+
+        assert_eq!(alice_balance, expected_alice_balance);
+
+        if tenure_id <= 1 {
+            if tenure_id < 1 {
+                // Alice has not locked up STX
+                let alice_balance = get_balance(&mut peer, &key_to_stacks_addr(&alice).into());
+                assert_eq!(alice_balance, 1024 * POX_THRESHOLD_STEPS_USTX);
+
+                let alice_account = get_account(&mut peer, &key_to_stacks_addr(&alice).into());
+                assert_eq!(
+                    alice_account.stx_balance.amount_unlocked(),
+                    1024 * POX_THRESHOLD_STEPS_USTX
+                );
+                assert_eq!(alice_account.stx_balance.amount_locked(), 0);
+                assert_eq!(alice_account.stx_balance.unlock_height(), 0);
+            }
+            let min_ustx = with_sortdb(&mut peer, |ref mut chainstate, ref sortdb| {
+                chainstate.get_stacking_minimum(sortdb, &tip_index_block)
+            })
+            .unwrap();
+            assert_eq!(min_ustx, total_liquid_ustx / 480);
+
+            // no reward addresses
+            let reward_addrs = with_sortdb(&mut peer, |ref mut chainstate, ref sortdb| {
+                get_reward_addresses_with_par_tip(chainstate, &burnchain, sortdb, &tip_index_block)
+            })
+            .unwrap();
+            assert_eq!(reward_addrs.len(), 0);
+
+            // record the first reward cycle when Alice's tokens get stacked
+            let tip_burn_block_height =
+                get_par_burn_block_height(peer.chainstate(), &tip_index_block);
+            alice_reward_cycle = 1 + burnchain
```

Comment:
I think this is going to get assigned twice, since `tenure_id <= 1` is true twice?  But, I don't think the value is different either way, since `1 + burnchain.block_height_to_reward_cycle(tip_burn_block_height)` will be the same in both cases?  Can we confirm?

Reply:
Yep -- it got assigned to twice, fixed in c46422e